Objective-C/Tests/QueryTest+Main.m

```diff
@@ -1873,5 +1794,280 @@ - (void) validateN1QLQuery: (NSString*)queryString {
+        NSArray<CBLQueryResult*>* rows = [change.results allObjects];
+        if (count == 1) {
+            
+            // initial notification about already existing result set!
+            AssertEqual(rows.count, 1);
+            AssertEqual([rows[0] integerAtIndex: 0], 7);
+        } else if (count == 2) {
+            
+            // newly added doc notification
+            AssertEqual(rows.count, 2);
+            AssertEqual([rows[0] integerAtIndex: 0], 3);
+            
+            [first fulfill];
+        }
+    }];
+    
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.5 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+        [self createDocNumbered: 3 of: 10];
+    });
+    
+    // wait for first set of notifications
+    [self waitForExpectations: @[first] timeout: 10.0];
+    AssertEqual(count, 2);
+    [q removeChangeListenerWithToken: token1];
+    
+    // adding a second listener is returning the last results!
+    XCTestExpectation* second = [self expectationWithDescription: @"second-notification"];
+    count = 0;
+    id token2 = [q addChangeListener: ^(CBLQueryChange* change) {
+        count++;
+        
+        // newly added doc notification!
+        NSArray<CBLQueryResult*>* rows = [change.results allObjects];
+        AssertEqual(rows.count, 2);
+        AssertEqual([rows[0] integerAtIndex: 0], 3);
+        
+        [second fulfill];
+    }];
+    
+    // wait for second notification
+    [self waitForExpectations: @[second] timeout: 5.0];
+    AssertEqual(count, 1);
+    [q removeChangeListenerWithToken: token2];
+}
+
+- (void) testLiveQueryReturnsEmptyResultSet {
+    [self createDocNumbered: 1 of: 10];
+    CBLQuery* q = [CBLQueryBuilder select: @[[CBLQuerySelectResult property: @"number1"]]
+                                     from: [CBLQueryDataSource database: self.db]
+                                    where: [[CBLQueryExpression property: @"number1"] lessThan: [CBLQueryExpression integer: 10]]
+                                  orderBy: @[[CBLQueryOrdering property: @"number1"]]];
+    
+    // first listener
+    XCTestExpectation* x = [self expectationWithDescription: @"live-query-finish"];
+    __block int count = 0;
+    id token = [q addChangeListener: ^(CBLQueryChange* change) {
+        count++;
+        NSArray<CBLQueryResult*>* rows = [change.results allObjects];
+        if (count == 1) {
+            // initial notification about already existing result set!
+            AssertEqual(rows.count, 1);
+            AssertEqual([rows[0] integerAtIndex: 0], 1);
+        } else if (count == 2) {
+            
+            // deleted the doc should return empty result set!
+            AssertEqual(rows.count, 0);
+            [x fulfill];
+        }
+    }];
+    
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.5 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+        NSError* error = nil;
+        Assert([self.db purgeDocumentWithID: @"doc1" error: &error]);
```

Comment:
This test can be simplified by just have the query returning empty result from the first notification and nothing else.